modules/text2vec-openai/ent/vectorization_result.go

```diff
@@ -37,30 +37,34 @@ func GetRateLimitsFromHeader(header http.Header) *modulecomponents.RateLimits {
 	remainingTokens := getHeaderInt(header, "x-ratelimit-remaining-tokens")
 
 	// azure returns 0 as limit, make sure this does not block anything by setting a high value
-	if limitTokens == 0 {
+	if isAzure {
 		limitTokens = dummyLimit
-	}
-	if limitRequests == 0 {
 		limitRequests = dummyLimit
 	}
+
+	updateWithMissingValues := false
+	if limitRequests < 0 || limitTokens < 0 || remainingRequests < 0 || remainingTokens < 0 {
+		updateWithMissingValues = true
```

Comment:
it would be very interesting to log all headers here (omitting the sensitive ones of course). We might discover that we are parsing stuff incorrectly or there is some additional "error" metric or something.